### ISSUE TITLE
ATTACH Fails With Foreign Key Constraint

### ISSUE THREAD

**Issue by ameijer** [duckdb/duckdb#9137]
### What happens?

When a database is built with a table that has a foreign key constraint, we don't seem to be able to attach it to another DB. 

### To Reproduce

```
$ duckdb
v0.8.1 6536a77232
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D .open sourcedb.duckdb
D CREATE TABLE parent(Id UUID PRIMARY KEY);
D CREATE TABLE child(ParentId UUID REFERENCES parent(Id));
<quit duckdb>

$ duckdb
v0.8.1 6536a77232
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D ATTACH 'sourcedb.duckdb' AS source;
Error: Binder Error: Catalog "source" does not exist!
```

If you re-create the experiment without the foreign key constraint, it works fine: 

```
$ duckdb
v0.8.1 6536a77232
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D .open sourcedb2.duckdb
D CREATE TABLE parent(Id UUID PRIMARY KEY);
D CREATE TABLE child(ParentId UUID);
<quit>

$ duckdb
v0.8.1 6536a77232
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D ATTACH 'source2.duckdb' AS source;
```

### OS:

OSX

### DuckDB Version:

v0.8.1 6536a77232

### DuckDB Client:

CLI

### Full Name:

Alex Meijer

### Affiliation:

Kubecost

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
Thanks for raising the issue!

We are already aware of this problem, it is on our radar :)

**Comment by szarnyasg** [comment]
This was fixed by #9932
